server/src/main/java/org/apache/iotdb/db/qp/executor/PlanExecutor.java

```diff
@@ -713,7 +713,9 @@ public void delete(DeletePlan deletePlan) throws QueryProcessException {
     try {
       Set<PartialPath> existingPaths = new HashSet<>();
       for (PartialPath p : deletePlan.getPaths()) {
-        existingPaths.addAll(getPathsName(p));
+        if (!getPathsName(p).isEmpty()) {
```

Comment:
It seems that you have get all the timeseries for each path here, there is no need for you to call it again somewhere else, you can pass it from here.